### PR TITLE
fix!: consistent split evaluations

### PR DESCRIPTION
**Pull request by khvn26**
Contributes to https://github.com/Flagsmith/flagsmith/issues/2186.

### CODE REVIEW

File: flagengine/segments/evaluator.go [hunk 1]
```diff
@@ -24,8 +24,14 @@ func EvaluateIdentityInSegment(
 		traits = overrideTraits
 	}
 
+	var identityID string
```
Comment (matthewelwell) [inline flagengine/segments/evaluator.go:27]: Can we use `identityHashKey` here?

File: flagengine/segments/evaluator_test.go [hunk 2]
```diff
@@ -297,7 +297,42 @@ func TestIdentityInSegmentPercentageSplit(t *testing.T) {
...
+}
+
+func TestIdentityInSegmentPercentageSplitUsesDjangoID(t *testing.T) {
+	cases := []struct {
+		identity       *identities.IdentityModel
+		expectedResult bool
+	}{
+		{&identities.IdentityModel{
+			DjangoID:          1,
+			Identifier:        "noDjangoID",
```
Comment (matthewelwell) [inline flagengine/segments/evaluator_test.go:310]:
```suggestion
			Identifier:        "withDjangoID",
```

File: flagengine/utils/hashing.go [hunk 3]
```diff
@@ -35,3 +35,7 @@ var hashedPercentageForObjectIdsFunc = getHashedPercentageForObjectIds
 func MockSetHashedPercentageForObjectIds(fn func([]string, int) float64) {
 	hashedPercentageForObjectIdsFunc = fn
 }
+
+func ResetMocks() {
```
Comment (matthewelwell) [inline flagengine/utils/hashing.go:39]: Perhaps this is a hangover from a previous attempt at mocking? I can't see that this is being used anywhere?

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: flagengine/utils/hashing.go (hunk 3 above)
Comment (khvn26) [reply flagengine/utils/hashing.go:39]: https://github.com/Flagsmith/flagsmith-go-client/pull/88#diff-e4c09beef6da84b10fa7aeaba7ceb03078ba6215f35c0ab95030d83a62fa5c70R300

Comment (khvn26) [reply flagengine/utils/hashing.go:39]: It's used here:  https://github.com/Flagsmith/flagsmith-go-client/blob/6ffd9c634841afae45ff6edbdccd8799351e2689/flagengine/segments/evaluator_test.go#L300

The previous code that was supposed to reset the function worked incorrectly, causing an infinite recursion.

File: flagengine/segments/evaluator.go (hunk 1 above)
Comment (khvn26) [reply flagengine/segments/evaluator.go]: Done.

File: flagengine/segments/evaluator_test.go (hunk 2 above)
Comment (matthewelwell) [reply flagengine/segments/evaluator_test.go:310]: Yes, this is true. It bothers me somewhat that we're creating an identity with an identifier of `"noDjangoID"`, for an identity that does have a `DjangoID` value - it looks like a mistake which could be confusing later. 

Can we choose another identifier that is consistent across the 2 identities?

Comment (khvn26) [reply flagengine/segments/evaluator_test.go]: Changed to "Test".

**Review by matthewelwell — APPROVED** [review]

**Comment by khvn26** [comment]
Waiting for @gagantrivedi to determine when exaclty the PR should be merged — before or after the v3 changes in #79.

File: flagengine/segments/evaluator.go [hunk 4]
```diff
@@ -24,8 +24,14 @@ func EvaluateIdentityInSegment(
 		traits = overrideTraits
 	}
 
+	var identityHashKey string
```
Comment (gagantrivedi) [inline flagengine/segments/evaluator.go:27]: Can we add the default value here, i.e: identity.CompositeKey() and remove the else?

**Review by gagantrivedi — CHANGES_REQUESTED** [review]

File: flagengine/segments/evaluator.go (hunk 4 above)
Comment (khvn26) [reply flagengine/segments/evaluator.go]: Not sure how I missed that leaving the assignment inside the cycle was meaningless. Also learned about the compiler inlining. Thanks for catching that!

**Review by gagantrivedi — APPROVED** [review]

**Pull request merged**